tests/integration/steps/deployment.go

```diff
@@ -329,6 +330,55 @@ func CreateServiceWithPort(ctx context.Context, appName, namespace string, port,
 	return ctx, err
 }
 
+func VerifyEnvVariableOnDeployment(ctx context.Context, envVarName, deploymentName, namespace, envVarVal string) error {
+	k8sClient, err := testcontext.GetK8sClientFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	return retry.Do(func() error {
+		var dep v1.Deployment
+		err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: deploymentName}, &dep)
+		if err != nil {
+			return err
+		}
+
+		for _, c := range dep.Spec.Template.Spec.Containers {
+			for _, e := range c.Env {
+				if e.Name == envVarName && e.Value == envVarVal {
+					return nil
+				}
+			}
+		}
+		return fmt.Errorf("expected env variable %s on Deployment %s with value %s", envVarName, deploymentName, envVarVal)
+	}, testcontext.GetRetryOpts()...)
+}
+
+func VerifyLackEnvVariableOnDeployment(ctx context.Context, envVarName, deploymentName, namespace string) error {
```

Comment:
Maybe a more expressive name is easier to understand `VerifyEnvVariableIsNotOnDeployment`?

Reply:
we can do that